### PR TITLE
FOUNDATIONS : Modify `Score`

### LINKED ISSUE #234: FOUNDATIONS: Modify `Score`

### PR DESCRIPTION
**Pull request by Xurshid-A-J**
Closes #234

### CODE REVIEW

File: Tarteeb.Api.Tests.Unit/Services/Foundations/Scores/ScoreServiceTests.Validations.Modify.cs [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+                broker.GetCurrentDateTime(), Times.Once);
+
+            this.loggingBrokerMock.Verify(broker =>
+                broker.LogError(It.Is(SameExceptionAs
+                    (expectedScoreValidationException))), Times.Once);
+
+            this.storageBrokerMock.Verify(broker =>
+                 broker.UpdateScoreAsync(It.IsAny<Score>()), Times.Never);
+
+            this.storageBrokerMock.VerifyNoOtherCalls();
```
Comment (Javlonbek-dev) [inline Tarteeb.Api.Tests.Unit/Services/Foundations/Scores/ScoreServiceTests.Validations.Modify.cs:130]: Ketma ketlikka amal qiling

Comment (Xurshid-A-J) [reply Tarteeb.Api.Tests.Unit/Services/Foundations/Scores/ScoreServiceTests.Validations.Modify.cs:130]: Resolved

File: Tarteeb.Api.Tests.Unit/Services/Foundations/Scores/ScoreServiceTests.Exceptions.Modify.cs [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+            SqlException sqlException = CreateSqlException();
+
+            var failedScoreStorageException =
+                new FailedScoreStorageException(sqlException);
+
+            var expectedScoreDependencyException =
+                new ScoreDependencyException(failedScoreStorageException);
+
+            this.dateTimeBrokerMock.Setup(broker =>
+               broker.GetCurrentDateTime()).Throws(sqlException);
```
Comment (GulchexrasGithub) [inline Tarteeb.Api.Tests.Unit/Services/Foundations/Scores/ScoreServiceTests.Exceptions.Modify.cs:37]: brokerdan avval tab kerak

Comment (Xurshid-A-J) [reply Tarteeb.Api.Tests.Unit/Services/Foundations/Scores/ScoreServiceTests.Exceptions.Modify.cs:37]: Resolved

File: Tarteeb.Api.Tests.Unit/Services/Foundations/Scores/ScoreServiceTests.Exceptions.Modify.cs [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+
+            // then
+            actualScoreDependencyException.Should().
+                BeEquivalentTo(expectedScoreDependencyException);
+
+            this.storageBrokerMock.Verify(broker =>
+                broker.SelectScoreByIdAsync(ScoreId), Times.Once());
+
+            this.dateTimeBrokerMock.Verify(broker =>
+                broker.GetCurrentDateTime(), Times.Once());
```
Comment (ElbekDeveloper) [inline Tarteeb.Api.Tests.Unit/Services/Foundations/Scores/ScoreServiceTests.Exceptions.Modify.cs:106]: Times.Once () kerak emas

File: Tarteeb.Api.Tests.Unit/Services/Foundations/Scores/ScoreServiceTests.Exceptions.Modify.cs [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+        [Fact]
+        public async Task ShouldThrowDependencyValidationExceptionOnModifyIfDatabaseUpdateConcurrencyErrorOccursAndLogItAsync()
+        {
+            // given 
+            int minutesInPast = GetRandomNegativeNumber();
+            DateTimeOffset randomDateTime = GetRandomDateTime();
+            Score randomScore = CreateRandomModifyScore(randomDateTime);
+            Score someScore = randomScore;
+            someScore.CreatedDate = randomDateTime.AddMinutes(minutesInPast);
+            Guid ScoreId = someScore.Id;
```
Comment (ElbekDeveloper) [inline Tarteeb.Api.Tests.Unit/Services/Foundations/Scores/ScoreServiceTests.Exceptions.Modify.cs:126]: ScoreId emas scoreId

File: Tarteeb.Api.Tests.Unit/Services/Foundations/Scores/ScoreServiceTests.Validations.Modify.cs [hunk 5]
```diff
@@ -0,0 +1,385 @@
...
+               key: nameof(Score.Grade),
+               values: "Grade is required");
+
+            invalidScoreException.AddData(
+               key: nameof(Score.Weight),
+               values: "Weight is required");
+
+            invalidScoreException.AddData(
+               key: nameof(Score.EffortLink),
+               values: "Text is required");
```
Comment (ElbekDeveloper) [inline Tarteeb.Api.Tests.Unit/Services/Foundations/Scores/ScoreServiceTests.Validations.Modify.cs:84]: Bu optional

File: Tarteeb.Api.Tests.Unit/Services/Foundations/Scores/ScoreServiceTests.Validations.Modify.cs [hunk 6]
```diff
@@ -0,0 +1,385 @@
...
+                 broker.UpdateScoreAsync(It.IsAny<Score>()), Times.Never);
+
+            this.storageBrokerMock.VerifyNoOtherCalls();
+            this.dateTimeBrokerMock.VerifyNoOtherCalls();
+            this.loggingBrokerMock.VerifyNoOtherCalls();
+        }
+
+        [Fact]
+        public async Task ShouldThrowValidationExceptionOnModifyIfUpdatedDateIsNotSameAsCreatedDateAndLogItAsync()
+        {
```
Comment (ElbekDeveloper) [inline Tarteeb.Api.Tests.Unit/Services/Foundations/Scores/ScoreServiceTests.Validations.Modify.cs:138]: ShouldThrowValidationExceptionOnModifyIfUpdatedDateIsSameAsCreatedDateAndLogItAsync

File: Tarteeb.Api.Tests.Unit/Services/Foundations/Scores/ScoreServiceTests.Exceptions.Modify.cs (hunk 4 above)
Comment (Xurshid-A-J) [reply Tarteeb.Api.Tests.Unit/Services/Foundations/Scores/ScoreServiceTests.Exceptions.Modify.cs:126]: Resolved

File: Tarteeb.Api.Tests.Unit/Services/Foundations/Scores/ScoreServiceTests.Validations.Modify.cs (hunk 5 above)
Comment (Xurshid-A-J) [reply Tarteeb.Api.Tests.Unit/Services/Foundations/Scores/ScoreServiceTests.Validations.Modify.cs:84]: Resolved

**Pull request merged**